evil-common.el

```diff
@@ -723,7 +724,7 @@ recursively."
            ((functionp cmd)
             (if (or (memq cmd '(digit-argument negative-argument))
                     (and found-prefix
-                         (memq cmd evil-digit-bound-motions)))
+                         (equal (vconcat seq) [48])))
```

Comment:
Why can't this just be:
```suggestion
                         (string= seq "0")))
```

Reply:
@tomdl89 `evil-extract-count` expects either a string or a vector. If you just use `string=`, it'll fail in cases like [this](https://github.com/emacs-evil/evil/blob/17c635f6e1f538bf4b2c3c276ddd9d4d165a52fb/evil-tests.el#L8617), since `string=` chokes on vectors.

To deal with this, my original solution was to deal with the string and vector cases separately:

```
((functionp cmd)
            (if (or (memq cmd '(digit-argument negative-argument))
                    (and found-prefix
                       (or (equal seq [48])
                            (equal seq "0")))) ;;use equal instead of string= here; if seq is a vector != [48], it'll choke
```

However, since any string can be represented by a vector (the converse is not true - see [M-right], above), we can just always convert `seq` to a vector with `vconcat` and compare it with the character code 48, and avoid having to add an extraneous `or` expression.

Reply:
OK makes sense. I think `(vector ?0)` or `(vconcat "0")` may be more readable than `[48]` but otherwise all good.